docs/usage/configuration/user_authentication/refresh_tokens.md

```diff
@@ -0,0 +1,124 @@
+
+There are some caveats:
+
+* If a third party gets both your access token and refresh token, they will be able to
+  continue to enjoy access to your session.
+  * This is still an improvement because you (the user) will notice when *your*
+    session expires and you're not able to use your refresh token.
+    That would be a giveaway that someone else has compromised your session.
+    You would be able to log in again and terminate that session.
+    Previously (with long-length access tokens), a third party that has your access
+    token could go undetected for a very long time.
+* Clients need to implement support for refresh tokens in order for them to be a
+  useful mechanism.
+  * It is up to homeserver administrators if they want to issue long-lived access
+    tokens to clients not implementing refresh tokens.
+    * For compatibility, it is likely that they should at least until client support
+      is widespread.
+      * Users with clients that support refresh tokens will still benefit from the
+        added security; it's not possible to downgrade a session to using long-lived
+        access tokens so this effectively gives users the choice.
+    * In a closed environment where all users use known clients, this may not be
+      an issue as the homeserver administrator can know if the clients have refresh
+      token support.
+
+
+## Configuration Guide
+
+The following configuration options, in the `registration` section, are related:
+
+* `session_lifetime`: maximum length of a session, even if it's refreshed.
+  In other words, the client must log in again after this time period.
+  In most cases, this can be unset (infinite) or set to a long time (years or months).
+* `refreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients supporting refresh tokens.
+  This should be short; a good value might be 5 minutes (`5m`).
+* `nonrefreshable_access_token_lifetime`: lifetime of access tokens that are created
+  by clients which don't support refresh tokens.
+  Make this short if you want to force use of refresh tokens.
+  Make this long if you don't want to inconvenience users of clients which don't
+  support refresh tokens.
+* `refresh_token_lifetime`: lifetime of refresh tokens.
+  Unless you want to log inactive sessions out, it is often fine to use a long
+  value here or even leave it unset (infinite).
+  Beware that making it too short will inconvenience clients that do not connect
+  very often, including mobile clients and clients of infrequent users.
+
+
+### Using refresh token expiry to log out inactive sessions
+
+If you'd like to force sessions to be logged out upon inactivity, you can enable
+refreshable access token expiry and refresh token expiry.
+
+Assuming that clients keep their access token valid whilst they are active, and
+that you set `refresh_token_lifetime` to some value *R* and
+`refreshable_access_token_lifetime` to some value *A* (which is not bigger than *R*),
```

Comment:
I parse this as `A <= R`. Judging by point 1 below, I think you don't want to allow `A = R`?

Reply:
There's nothing too wrong with `A = R`; clients would just have to be diligent about refreshing...

The only confusing part is that the section below explaining the range of inactivities that will kick a client would then have a lower bound of zero.
In reality this depends on client behaviour and it's unlikely that clients would wait until the absolute last second to refresh their tokens, but I was trying to be helpful to admins who are thinking something like 'I want sessions inactive for 15 minutes to be logged out' and trying to paint a picture that it depends on when the clients happen to have refreshed their token — are you really OK with that meaning that an inactivity of (e.g. 5 minutes) could also trigger that behaviour?